src/radical/pilot/pilot_description.py

```diff
@@ -257,6 +262,14 @@ def _verify(self):
         if not self.get('resource'):
             raise ValueError("Pilot description needs 'resource'")
 
+        if self.get('nodes'):
+
+            if self.get('cores'):
+                raise ValueError("Pilot description needs 'cores' *or* 'nodes'")
+
+            if self.get('gpus'):
+                raise ValueError("Pilot description needs 'cores' *or* 'nodes'")
```

Comment:
just a side note - maybe combine it? -> `if self.get('cores') or self.get('gpus'):`
p.s. in gpus-part, error message points to `cores`